pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -521,20 +519,16 @@ func (r *Builder) mapDisks(vm *model.Workload, persistentVolumeClaims []core.Per
 		bus = VirtioBus
 	}
 
-	for i := range persistentVolumeClaims {
-		pvc := &persistentVolumeClaims[i]
-		pvcMap[pvc.Annotations[AnnImportDiskId]] = pvc
-	}
-	for i, av := range vm.Volumes {
-		image := &model.Image{}
-		err := r.Source.Inventory.Find(image, ref.Ref{Name: fmt.Sprintf("%s-%s", r.Migration.Name, av.ID)})
+	for _, pvc := range persistentVolumeClaims {
+		image, err := r.getImageFromPVC(&pvc)
 		if err != nil {
+			r.Log.Error(err, "image not found in inventory", "imageID", pvc.Name)
 			return
 		}
-		pvc := pvcMap[av.ID]
-		volumeName := fmt.Sprintf("vol-%v", i)
-		volume := cnv.Volume{
-			Name: volumeName,
+
+		cnvVolumeName := fmt.Sprintf("disk-%v", pvc.Annotations[AnnImportDiskId])
```

Comment:
is it safe to change it from vol-%v to disk-%v at this point? I think we rely on it only when creating the vm but trying to think what happens when the controller is upgraded while there's an ongoing migration

Reply:
I will move it back to `vol`